### PR TITLE
[Ed25519] Use BouncyCastle instead of Chaos.NaCl

### PR DESCRIPTION
**Pull request by scott-xu**

### CODE REVIEW

**Comment by WojciechNagorski** [comment]
Can we remove https://github.com/sshnet/SSH.NET/blob/develop/src/Renci.SshNet/Security/Chaos.NaCl/Ed25519.cs in this PR?

**Comment by scott-xu** [comment]
Was going to remove the whole Chaos.NaCl folder once several other PRs get merged. Do you want me to remove it piece by piece?

File: src/Renci.SshNet/Security/Cryptography/ED25519Key.cs [hunk 1]
```diff
@@ -91,8 +92,7 @@ public ED25519Key(SshKeyData publicKeyData)
                 throw new ArgumentException($"Invalid Ed25519 public key data ({publicKeyData.Name}, {publicKeyData.Keys.Length}).", nameof(publicKeyData));
             }
 
-            PublicKey = publicKeyData.Keys[0].ToByteArray().Reverse().TrimLeadingZeros().Pad(Ed25519.PublicKeySizeInBytes);
-            PrivateKey = new byte[Ed25519.ExpandedPrivateKeySizeInBytes];
+            PublicKey = publicKeyData.Keys[0].ToByteArray().Reverse().TrimLeadingZeros().Pad(Ed25519.PublicKeySize);
```
Comment (Rob-Hague) [inline src/Renci.SshNet/Security/Cryptography/ED25519Key.cs:95]: I'd just keep the PrivateKey not-null unless we enable nullability here

```suggestion
            PublicKey = publicKeyData.Keys[0].ToByteArray().Reverse().TrimLeadingZeros().Pad(Ed25519.PublicKeySize);
            PrivateKey = new byte[Ed25519.SecretKeySize];
```

**Review by Rob-Hague — APPROVED** [review]

**Comment by Rob-Hague** [comment]
> Was going to remove the whole Chaos.NaCl folder once several other PRs get merged. Do you want me to remove it piece by piece?

I think either way is fine

**Pull request merged**